test/hotel-search/Client.spec.js

```diff
@@ -330,6 +330,42 @@ describe("HotelSearchClient", function() {
 
       expect(client.merger.__staticData.brands[1]).to.equal(brand);
     });
+
+    it("no hotel can have more than 1 rate from a provider when search's status is not'done'", () =>{
+      var response = {
+        done: false,
+        hotels: [{ id: 1},{ id: 2}],
+        rates: [
+          { id: 1, hotelId: 1, providerCode: "a.com", price: {amount: 100, taxAmountUsd: 1} },
+          { id: 2, hotelId: 1, providerCode: "a.com", price: {amount: 110, taxAmountUsd: 1} },
+          { id: 3, hotelId: 2, providerCode: "a.com", price: {amount: 120, taxAmountUsd: 1} },
+          { id: 4, hotelId: 2, providerCode: "b.com", price: {amount: 130, taxAmountUsd: 1} },
+          { id: 5, hotelId: 1, providerCode: "a.com", price: {amount: 140, taxAmountUsd: 1} },
+        ]
+      }
+      client.handleSearchResponse(response);
+      expect(client.merger.__hotelMap[1].rates.length).to.equal(1);
+      expect(client.merger.__hotelMap[2].rates.length).to.equal(2);
+      expect(client.merger.__hotelMap[2].rates[0].providerCode).to.not.equal(
+        client.merger.__hotelMap[2].rates[1].providerCode);
+    });
+
+    it("some hotels may have more than 1 from a provider rate when search's status is 'done'", () =>{
```

Comment:
should be `some hotels may have more than 1 rate from a provider  when search's status is 'done'` ?